test/bundler/bun-build-api.test.ts

```diff
@@ -250,19 +250,6 @@ describe("Bun.build", () => {
     expect(x.logs[0].position).toEqual(null);
   });
 
-  test("warnings do not fail a build", async () => {
```

Comment:
why was this test removed? it will hide a potential regression if this warning or syntax is a failing build.

we should still print the warning because `import { createElement } from "react";` *is still* the classic JSX runtime and the warning points out a probably mistake in the user's code.

also, this warning seems to only be printed once per build

Reply:
Put the key after the spread props is a valid approach, and I suggest not warning for it(ts/esbuild/swc also do not warn it).

Reply:
Additionally, this warning will break transpiling, eg.

```js
const transpiler = new Bun.Transpiler({ loader: "jsx" });
try {
  const result = await transpiler.transform(`console.log(<div {...obj} key="after" />, <div key="before" {...obj} />)`);
  console.log(result);
} catch (e) {
  console.log("err", e);
}
```

```
err 1 | console.log(<div {...obj} key="after" />, <div key="before" {...obj} />)
                                                                    ^
warn: "key" prop before a {...spread} is deprecated in JSX. Falling back to classic runtime.
   at input.jsx:1:65
```

Reply:
didnt see this. the warning is valuable, but `Bun.Transpiler` shouldnt throw on it. really any warning shouldnt throw. jarred suggested we make the default logLevel for transpiler be `err` instead of `warn`, but it seems this isnt too easily configurable.

easy fix would just be to make it never throw on warnings in JSTranspiler.zig

```diff
-if ((this.bundler.log.warnings + this.bundler.log.errors) > 0) {
+if (this.bundler.log.errors > 0) {
   globalThis.throwValue(this.bundler.log.toJS(globalThis, globalThis.allocator(), "Parse error"));
   return .zero;
 }
```

maybe this behavior is desired sometimes and we should allow passing a logLevel property to bun.Transpiler

Reply:
I see, I will add the warning back.
Let me clarify my understanding to avoid confusion: We should issue a warning when the key appears **after** spread props, not **before**, is that correct?

Reply:
Awww… this is a bit confusing for me. You see, the `createElement` downgrade is for the key **after** spread props, but should we issue a warning for the non-downgraded key **before** spread props? What should the content of this warning message be?

Reply:
BTW, `react-jsx` can handle key before spread perfectly, ([see](https://www.typescriptlang.org/play?target=99&jsx=4#code/MYewdgziA2CmB00QHMAUAeAJgSwG4AIBveEkAIwCsBffAa1gE8BeAIgEMAzAF1gCcX8AegB8AGnxY8dRqzKwOIXrAHFSlGiICUQA))

Reply:
i double checked esbuild and it behaves how you have modified it.

so before = no warn, automatic jsx runtime
after = warn, fallback to classic

and this fixes the output to match tsc and esbuild